[upgrade] Use modern SASS implementation

domq opened the pull request:
- `node-sass` → `sass`
- Make minimum amount of fixes to go back to 0 errors, and about the same number of warnings as previously

Tested with `meld` between old and new CSS, as well as eyeballing the elements in the reader app.

domq committed:
[upgrade] node-sass → sass

```
yarn remove node-sass
yarn add -D sass
```

domq committed:
[fix] `outputStyle`: `'uncompressed'` → `'expanded'`

As seen in https://stackoverflow.com/a/57904134

domq committed:
[fix] any SASS errors that just cropped up

domq committed:
[fix] Deprecated uses of the division operator in “our” code

domq committed:
[fix] Silence warnings caused by Bootstrap's SCSS code

- Enable `{ sassOptions: { quietDeps: true } }` for `sass-loader`
- In order for that to have any effect, `sass` needs to know the difference between “our” SCSS code and “their” SCSS code. To this end, stop using relative imports (`../../node_modules/`) to import the bits and pieces of Bootstrap we use.

williambelle commented on assets/components/molecules/question/question.scss:
```diff
@@ -77,7 +78,7 @@ $animation: 0.2s;
   // checked animation and styles
   .custom-control-input:checked + .custom-control-label {
-    @extend .custom-control-label:hover;
+    @extend .custom-control-label, :hover;
```
I'm not sure about this part.

| Before  | After  | 
|---|---|
|  ![Screenshot from 2022-11-03 09-14-53](https://user-images.githubusercontent.com/2843501/199675886-ec7073cd-2bd5-4a75-9d67-550194b69a81.png) |  ![Screenshot from 2022-11-03 09-16-11](https://user-images.githubusercontent.com/2843501/199675919-b7340210-e611-4663-b8f5-3ee269fd7d42.png) | 

http://localhost:3000/#/molecules/question

domq replied:
Nice catch! Need to put on my thinking hat.

domq replied:
Yep, that was wrong. Fixed in the last pushed version, thanks for spotting.

domq committed:
[refactor] Extract `:hover` behavior from `trapeze-span-*-container` mixins

This lets us call them without using `@extends` (e.g. for selected questions).

domq committed:
[fix] selected question rendering (as per code review)

- As documented in https://sass-lang.com/documentation/breaking-changes/extend-compound , we want to stop using `@extends` altogether
- This almost suffices, but then the `:hover`-like behavior of a selected question (i.e. the fact that the question stays raised / pushed) disappears. Bring it back by invoking the newly-introduced `trapeze-span-*-container-active` mixins from within the appropriate media queries.
